synapse/storage/databases/main/devices.py

```diff
@@ -1062,16 +1062,30 @@ async def get_user_ids_requiring_device_list_resync(
 
         return {row["user_id"] for row in rows}
 
-    async def mark_remote_user_device_cache_as_stale(self, user_id: str) -> None:
+    async def mark_remote_users_device_caches_as_stale(
+        self, user_ids: StrCollection
+    ) -> None:
         """Records that the server has reason to believe the cache of the devices
         for the remote users is out of date.
         """
-        await self.db_pool.simple_upsert(
-            table="device_lists_remote_resync",
-            keyvalues={"user_id": user_id},
-            values={},
-            insertion_values={"added_ts": self._clock.time_msec()},
-            desc="mark_remote_user_device_cache_as_stale",
+
+        def _mark_remote_users_device_caches_as_stale_txn(
+            txn: LoggingTransaction,
+        ) -> None:
+            # TODO add insertion_values support to simple_upsert_many and use
+            #      that!
```

Comment:
Could likely replace this with raw SQL if we wanted.

Reply:
(I think I wanted to use this somewhere in the receipts or event push actions code, but can't seem to find where.)

Reply:
I'm on the fence about raw SQL vs just leaving it like this.
It'd be interesting to have support for pipelined statements (c.f. https://www.psycopg.org/psycopg3/docs/advanced/pipeline.html) in the future. I do generally like the readability of our helpers...